--- at-spi2-atk-2.2.1/configure.ac-orig	2011-10-24 12:21:46.023071995 -0500
+++ at-spi2-atk-2.2.1/configure.ac	2011-10-24 12:21:53.847105319 -0500
@@ -20,7 +20,7 @@ AC_SUBST(LT_CURRENT)
 AC_SUBST(LT_REVISION)
 AC_SUBST(LT_AGE)
 
-AM_INIT_AUTOMAKE([-Wall foreign no-dist-gzip dist-xz])
+AM_INIT_AUTOMAKE([-Wall foreign])
 
 dnl translation of any at-spi strings: used only for .server
 IT_PROG_INTLTOOL([0.40.0])
