--- a/m4/gecko.m4
+++ b/m4/gecko.m4
@@ -113,7 +116,7 @@
 
 if $PKG_CONFIG --exists  ${gecko_cv_gecko}-xpcom; then
 	_GECKO_INCLUDE_ROOT="`$PKG_CONFIG --variable=includedir ${gecko_cv_gecko}-xpcom`"
-	_GECKO_CFLAGS="-I$_GECKO_INCLUDE_ROOT"
+	_GECKO_CFLAGS="`$PKG_CONFIG --cflags ${gecko_cv_gecko}-xpcom`"
 	_GECKO_LIBDIR="`$PKG_CONFIG --variable=libdir ${gecko_cv_gecko}-xpcom`"
 	_GECKO_HOME="`$PKG_CONFIG --variable=libdir ${gecko_cv_gecko}-xpcom`"
 	_GECKO_PREFIX="`$PKG_CONFIG --variable=prefix ${gecko_cv_gecko}-xpcom`"
@@ -308,8 +311,10 @@
 	version = "1.9";
 } else if (strncmp (MOZILLA_VERSION, "1.8", strlen ("1.8")) == 0) {
 	version = "1.8";
-} else {
+} else if (strncmp (MOZILLA_VERSION, "1.", strlen ("1.")) == 0) {
 	version = "1.7";
+} else {
+	version = "2.0";
 }
 #endif
 fprintf (stream, "%s\n", version);
@@ -328,7 +333,7 @@
 
 gecko_cv_gecko_version_int="$(echo "$gecko_cv_gecko_version" | $AWK -F . '{print [$]1 * 1000000 + [$]2 * 1000 + [$]3}')"
 
-if test "$gecko_cv_gecko_version_int" -lt "1007000" -o "$gecko_cv_gecko_version_int" -gt "1009000"; then
+if test "$gecko_cv_gecko_version_int" -lt "1007000" -o "$gecko_cv_gecko_version_int" -gt "2000000"; then
 	AC_MSG_ERROR([Gecko version $gecko_cv_gecko_version is not supported!])
 fi
 
@@ -345,6 +347,10 @@
 	AC_DEFINE([HAVE_GECKO_1_9],[1],[Define if we have gecko 1.9])
 	gecko_cv_have_gecko_1_9=yes
 fi
+if test "$gecko_cv_gecko_version_int" -ge "2000000"; then
+	AC_DEFINE([HAVE_GECKO_2_0],[1],[Define if we have gecko 2.0])
+	gecko_cv_have_gecko_2_0=yes
+fi
 
 fi # if gecko_cv_have_gecko
 
