diff -ruN evolution-data-server-2.30.1.orig/camel/camel-sasl-gssapi.c evolution-data-server-2.30.1/camel/camel-sasl-gssapi.c
--- evolution-data-server-2.30.1.orig/camel/camel-sasl-gssapi.c	2010-04-26 20:00:55.232977746 +0100
+++ evolution-data-server-2.30.1/camel/camel-sasl-gssapi.c	2010-04-26 20:04:04.341377459 +0100
@@ -31,18 +31,7 @@
 #include <string.h>
 #include <sys/socket.h>
 #include <sys/types.h>
-#ifdef HAVE_HEIMDAL_KRB5
-#include <krb5.h>
-#else
-#include <krb5/krb5.h>
-#endif
-#ifdef HAVE_ET_COM_ERR_H
-#include <et/com_err.h>
-#else
-#ifdef HAVE_COM_ERR_H
-#include <com_err.h>
-#endif
-#endif
+#include <kerberosv5/krb5.h>
 #ifdef HAVE_MIT_KRB5
 #include <gssapi/gssapi.h>
 #include <gssapi/gssapi_generic.h>
