--- a/js/src/jsgc.h	Tue Nov 15 16:22:47 2011 +0800
+++ b/js/src/jsgc.h	Tue Nov 15 16:27:01 2011 +0800
@@ -90,7 +90,11 @@
  */
 const size_t MAX_BACKGROUND_FINALIZE_KINDS = FINALIZE_LIMIT - (FINALIZE_OBJECT_LAST + 1) / 2;
 
+#if defined(SOLARIS) && (defined(__sparc) || defined(__sparcv9))
+const size_t ArenaShift = 13;
+#else
 const size_t ArenaShift = 12;
+#endif
 const size_t ArenaSize = size_t(1) << ArenaShift;
 const size_t ArenaMask = ArenaSize - 1;
 
