--- a/configure	2011-11-15 16:58:56.774289619 +0800
+++ b/configure	2011-11-15 16:59:12.689288454 +0800
@@ -5930,8 +5930,8 @@
 MOZ_PNG_CFLAGS=
 MOZ_PNG_LIBS='$(call EXPAND_LIBNAME_PATH,mozpng,$(DEPTH)/modules/libimg/png)'
 
-MOZ_JS_STATIC_LIBS='$(call EXPAND_LIBNAME_PATH,js_static,$(LIBXUL_DIST)/lib)'
-MOZ_JS_SHARED_LIBS='$(call EXPAND_LIBNAME_PATH,mozjs,$(LIBXUL_DIST)/lib)'
+MOZ_JS_STATIC_LIBS='$(call EXPAND_LIBNAME_PATH,js_static,$(DEPTH)/js/src)'
+MOZ_JS_SHARED_LIBS='$(LIBXUL_DIST)/lib/$(DLL_PREFIX)mozjs$(DLL_SUFFIX)'
 DYNAMIC_XPCOM_LIBS='-L$(LIBXUL_DIST)/bin -lxpcom -lxpcom_core -lmozalloc'
 MOZ_FIX_LINK_PATHS='-Wl,-rpath-link,$(LIBXUL_DIST)/bin -Wl,-rpath-link,$(prefix)/lib'
 XPCOM_FROZEN_LDOPTS='-L$(LIBXUL_DIST)/bin -lxpcom -lmozalloc'
diff --git a/js/src/Makefile.in b/js/src/Makefile.in
--- a/js/src/Makefile.in
+++ b/js/src/Makefile.in
@@ -123,7 +123,6 @@
 FORCE_SHARED_LIB = 1
 endif
 FORCE_STATIC_LIB = 1
-DIST_INSTALL = 1
 
 VPATH		= $(srcdir)
 
