--- a/netwerk/wifi/nsWifiScannerSolaris.cpp	Thu Jul  7 16:38:41 2011
--- b/netwerk/wifi/nsWifiScannerSolaris.cpp	Thu Jul  7 16:38:41 2011
@@ -174,7 +174,7 @@
     nsCOMArray<nsIWifiListener> currentListeners;
 
     {
-      MonitorAutoEnter mon(mMonitor);
+      ReentrantMonitorAutoEnter mon(mReentrantMonitor);
 
       for (PRUint32 i = 0; i < mListeners.Length(); i++) {
         if (!mListeners[i].mHasSentData || accessPointsChanged) {
@@ -223,7 +223,7 @@
 
     LOG(("waiting on monitor\n"));
 
-    MonitorAutoEnter mon(mMonitor);
+    ReentrantMonitorAutoEnter mon(mReentrantMonitor);
     mon.Wait(PR_SecondsToInterval(60));
   }
 
