--- a/browser/app/profile/firefox.js	2012-03-14 08:59:31.244537010 +0800
+++ b/browser/app/profile/firefox.js	2012-03-14 08:59:40.185001826 +0800
@@ -88,7 +88,7 @@
 // Disable add-ons installed into the shared user and shared system areas by
 // default. This does not include the application directory. See the SCOPE
 // constants in AddonManager.jsm for values to use here
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 11);
 
 // Dictionary download preference
 pref("browser.dictionaries.download.url", "https://addons.mozilla.org/%LOCALE%/firefox/dictionaries/");
