--- a/configure	Mon Jul 11 11:52:00 2011
+++ b/configure	Mon Jul 11 11:52:42 2011
@@ -19056,7 +19056,7 @@
 # Check whether --enable-update-channel or --disable-update-channel was given.
 if test "${enable_update_channel+set}" = set; then
   enableval="$enable_update_channel"
-  MOZ_UPDATE_CHANNEL=`echo $enableval | tr A-Z a-z`
+  MOZ_UPDATE_CHANNEL=`echo $enableval | tr -s '[:upper:]' '[:lower:]'`
 fi
 
 
@@ -21655,7 +21655,7 @@
 # Check whether --enable-chrome-format or --disable-chrome-format was given.
 if test "${enable_chrome_format+set}" = set; then
   enableval="$enable_chrome_format"
-  MOZ_CHROME_FILE_FORMAT=`echo $enableval | tr A-Z a-z`
+  MOZ_CHROME_FILE_FORMAT=`echo $enableval | tr -s '[:upper:]' '[:lower:]'`
 fi
 
 
@@ -24613,7 +24613,7 @@
 # "Profile" field, which controls profile location.
 
 if test -z "$MOZ_APP_NAME"; then
-   MOZ_APP_NAME=`echo $MOZ_APP_BASENAME | tr A-Z a-z`
+   MOZ_APP_NAME=`echo $MOZ_APP_BASENAME | tr -s '[:upper:]' '[:lower:]'`
 fi
 
 
