diff --git a/js/src/jsfun.h b/js/src/jsfun.h
index a52092c..9555a03 100644
--- a/js/src/jsfun.h
+++ b/js/src/jsfun.h
@@ -144,6 +144,7 @@
         void            *nativeOrScript;
     } u;
     JSAtom          *atom;        /* name for diagnostics and decompiling */
+    void            *dummy;       /* for building with Solaris Studio */
 
     bool optimizedClosure()  const { return FUN_KIND(this) > JSFUN_INTERPRETED; }
     bool isInterpreted()     const { return FUN_INTERPRETED(this); }
