--- gnome-settings-daemon-3.1.3/plugins/Makefile.am-orig	2011-07-06 22:46:49.721721914 -0500
+++ gnome-settings-daemon-3.1.3/plugins/Makefile.am	2011-07-06 22:46:54.732517690 -0500
@@ -6,7 +6,6 @@ enabled_plugins =	\
 	automount	\
 	background	\
 	clipboard	\
-	color		\
 	cursor		\
 	datetime	\
 	dummy		\
--- gnome-settings-daemon-3.1.3/configure.ac-orig	2011-07-06 22:49:24.159078844 -0500
+++ gnome-settings-daemon-3.1.3/configure.ac	2011-07-06 22:49:39.465472031 -0500
@@ -21,7 +21,6 @@ AC_STDC_HEADERS
 AC_PROG_CXX
 AM_PROG_CC_C_O
 AC_PROG_LIBTOOL
-AC_PATH_PNPIDS
 
 AC_HEADER_STDC
 
@@ -115,11 +114,6 @@ dnl ------------------------------------
 PKG_CHECK_MODULES(GNOME_DESKTOP, gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION)
 
 dnl ---------------------------------------------------------------------------
-dnl - Check for colord
-dnl ---------------------------------------------------------------------------
-PKG_CHECK_MODULES(COLORD, colord >= 0.1.12)
-
-dnl ---------------------------------------------------------------------------
 dnl - Check for libcanberra
 dnl ---------------------------------------------------------------------------
 PKG_CHECK_MODULES(LIBCANBERRA, libcanberra-gtk3,
@@ -128,15 +122,15 @@ PKG_CHECK_MODULES(LIBCANBERRA, libcanber
                   [have_libcanberra=false])
 AM_CONDITIONAL(HAVE_LIBCANBERRA, test "x$have_libcanberra" = "xtrue")
 
-dnl ---------------------------------------------------------------------------
-dnl - Check for LCMS2
-dnl ---------------------------------------------------------------------------
-PKG_CHECK_MODULES(LCMS, lcms2 >= 2.2, have_new_lcms=yes, have_new_lcms=no)
-if test x$have_new_lcms = xyes; then
-        AC_DEFINE(HAVE_NEW_LCMS,1,[Got new lcms2])
-else
-        PKG_CHECK_MODULES(LCMS, lcms2)
-fi
+#dnl ---------------------------------------------------------------------------
+#dnl - Check for LCMS2
+#dnl ---------------------------------------------------------------------------
+#PKG_CHECK_MODULES(LCMS, lcms2 >= 2.2, have_new_lcms=yes, have_new_lcms=no)
+#if test x$have_new_lcms = xyes; then
+#        AC_DEFINE(HAVE_NEW_LCMS,1,[Got new lcms2])
+#else
+#        PKG_CHECK_MODULES(LCMS, lcms2)
+#fi
 
 dnl ---------------------------------------------------------------------------
 dnl - Check for libnotify
@@ -623,5 +617,4 @@ ${NSS_DATABASE:+\
         System nssdb:             ${NSS_DATABASE}
 }\
         Profiling support:        ${enable_profiling}
-        pnp.ids filename:         ${ac_with_pnpids}
 "
