--- gnome-settings-daemon-3.4.1/configure.ac-orig	2012-05-09 15:46:48.200471749 -0500
+++ gnome-settings-daemon-3.4.1/configure.ac	2012-05-09 16:05:00.905688064 -0500
@@ -103,15 +103,15 @@ dnl - Check for gnome-desktop
 dnl ---------------------------------------------------------------------------
 PKG_CHECK_MODULES(GNOME_DESKTOP, gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION)
 
-dnl ---------------------------------------------------------------------------
-dnl - Check for LCMS2
-dnl ---------------------------------------------------------------------------
-PKG_CHECK_MODULES(LCMS, lcms2 >= 2.2, have_new_lcms=yes, have_new_lcms=no)
-if test x$have_new_lcms = xyes; then
-        AC_DEFINE(HAVE_NEW_LCMS,1,[Got new lcms2])
-else
-        PKG_CHECK_MODULES(LCMS, lcms2)
-fi
+#dnl ---------------------------------------------------------------------------
+#dnl - Check for LCMS2
+#dnl ---------------------------------------------------------------------------
+#PKG_CHECK_MODULES(LCMS, lcms2 >= 2.2, have_new_lcms=yes, have_new_lcms=no)
+#if test x$have_new_lcms = xyes; then
+#        AC_DEFINE(HAVE_NEW_LCMS,1,[Got new lcms2])
+#else
+#        PKG_CHECK_MODULES(LCMS, lcms2)
+#fi
 
 dnl ---------------------------------------------------------------------------
 dnl - Check for libnotify
@@ -200,17 +200,17 @@ dnl ------------------------------------
 
 PKG_CHECK_MODULES(MEDIA_KEYS, [gio-unix-2.0 libpulse >= $PA_REQUIRED_VERSION $GUDEV_PKG libpulse-mainloop-glib >= $PA_REQUIRED_VERSION libcanberra])
 
-dnl ---------------------------------------------------------------------------
-dnl - color
-dnl ---------------------------------------------------------------------------
-
-PKG_CHECK_MODULES(COLOR, [colord >= 0.1.9 gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION libcanberra-gtk3])
+#dnl ---------------------------------------------------------------------------
+#dnl - color
+#dnl ---------------------------------------------------------------------------
+#
+#PKG_CHECK_MODULES(COLOR, [colord >= 0.1.9 gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION libcanberra-gtk3])
 
 dnl ---------------------------------------------------------------------------
 dnl - wacom
 dnl ---------------------------------------------------------------------------
 
-PKG_CHECK_MODULES(WACOM, [libwacom >= $LIBWACOM_REQUIRED_VERSION x11 xi xtst $GUDEV_PKG gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION xorg-wacom])
+#PKG_CHECK_MODULES(WACOM, [libwacom >= $LIBWACOM_REQUIRED_VERSION x11 xi xtst $GUDEV_PKG gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION xorg-wacom])
 
 dnl ==============================================
 dnl PackageKit section
@@ -292,7 +292,18 @@ dnl ====================================
 # ---------------------------------------------------------------------------
 # UPower
 # ---------------------------------------------------------------------------
-PKG_CHECK_MODULES(UPOWER, upower-glib >= $UPOWER_REQUIRED_VERSION libcanberra-gtk3)
+PKG_CHECK_MODULES(UPOWER_GLIB,
+        upower-glib >= $UPOWER_REQUIRED_VERSION libcanberra-gtk3,
+        have_upower_glib=yes,
+        have_upower_glib=no)
+if test "x$have_upower_glib" = "xyes" ; then
+  AC_DEFINE(HAVE_UPOWER_GLIB, [], [Define if we have upower_glib])
+fi
+AM_CONDITIONAL(HAVE_UPOWER_GLIB, test x$have_upower_glib = xyes)
+AC_SUBST(HAVE_UPOWER_GLIB)
+AC_SUBST(UPOWER_GLIB_CFLAGS)
+AC_SUBST(UPOWER_GLIB_LIBS)
+
 if test x$enable_gudev != xno; then
 	PKG_CHECK_MODULES(BACKLIGHT_HELPER,
 		glib-2.0 >= $GLIB_REQUIRED_VERSION
@@ -512,7 +523,6 @@ echo "
         Maintainer mode:          ${USE_MAINTAINER_MODE}
 
         Session tracking:         ${SESSION_TRACKING}
-        LCMS DICT support:        ${have_new_lcms}
 
         Libnotify support:        ${have_libnotify}
         PackageKit support:       ${have_packagekit}
--- gnome-settings-daemon-3.4.1/plugins/Makefile.am-orig	2012-05-09 15:46:53.739080021 -0500
+++ gnome-settings-daemon-3.4.1/plugins/Makefile.am	2012-05-09 15:59:49.467860816 -0500
@@ -6,16 +6,13 @@ enabled_plugins =	\
 	automount	\
 	background	\
 	clipboard	\
-	color		\
 	cursor		\
 	dummy		\
-	power		\
 	housekeeping    \
 	keyboard	\
 	media-keys	\
 	mouse		\
 	sound		\
-	wacom		\
 	xrandr		\
 	xsettings	\
 	$(NULL)
--- gnome-settings-daemon-3.4.1/plugins/print-notifications/Makefile.am-orig	2012-05-09 16:20:43.689208890 -0500
+++ gnome-settings-daemon-3.4.1/plugins/print-notifications/Makefile.am	2012-05-09 16:22:25.243764824 -0500
@@ -25,7 +25,7 @@ libprint_notifications_la_LDFLAGS = 		\
 
 libprint_notifications_la_LIBADD  = 		\
 	$(CUPS_LIBS)			\
-	$(SETTINGS_PLUGIN_LIBS)
+	$(SETTINGS_PLUGIN_LIBS) $(LIBNOTIFY_LIBS)
 
 plugin_in_files = 		\
 	print-notifications.gnome-settings-plugin.in
--- gnome-settings-daemon-3.4.1/plugins/xrandr/gsd-xrandr-manager.c-orig	2012-05-09 16:18:37.665859617 -0500
+++ gnome-settings-daemon-3.4.1/plugins/xrandr/gsd-xrandr-manager.c	2012-05-09 16:19:53.534620761 -0500
@@ -38,7 +38,9 @@
 #include <gdk/gdk.h>
 #include <gdk/gdkx.h>
 #include <gtk/gtk.h>
+#if HAVE_UPOWER_GLIB
 #include <libupower-glib/upower.h>
+#endif
 #include <X11/Xatom.h>
 
 #define GNOME_DESKTOP_USE_UNSTABLE_API
@@ -104,7 +106,9 @@ struct GsdXrandrManagerPrivate
         GnomeRRScreen *rw_screen;
         gboolean running;
 
+#if HAVE_UPOWER_GLIB
         UpClient *upower_client;
+#endif
         gboolean laptop_lid_is_closed;
 
         GSettings       *settings;
@@ -461,8 +465,10 @@ apply_configuration_from_filename (GsdXr
                 }
         }
 
+#if HAVE_UPOWER_GLIB
         if (up_client_get_lid_is_closed (priv->upower_client))
                 turn_off_laptop_display_in_configuration (priv->rw_screen, config);
+#endif
 
         gnome_rr_config_ensure_primary (config);
 	success = gnome_rr_config_apply_with_time (config, priv->rw_screen, timestamp, error);
@@ -850,7 +856,11 @@ config_is_all_off (GnomeRRConfig *config
 static gboolean
 laptop_lid_is_closed (GsdXrandrManager *manager)
 {
+#if HAVE_UPOWER_GLIB
         return up_client_get_lid_is_closed (manager->priv->upower_client);
+#else
+        return FALSE;
+#endif
 }
 
 static gboolean
@@ -1927,6 +1937,7 @@ turn_off_laptop_display (GsdXrandrManage
         g_object_unref (config);
 }
 
+#if HAVE_UPOWER_GLIB
 static void
 power_client_changed_cb (UpClient *client, gpointer data)
 {
@@ -1953,7 +1964,7 @@ power_client_changed_cb (UpClient *clien
                         use_stored_configuration_or_auto_configure_outputs (manager, GDK_CURRENT_TIME);
         }
 }
-
+#endif
 gboolean
 gsd_xrandr_manager_start (GsdXrandrManager *manager,
                           GError          **error)
@@ -1976,10 +1987,12 @@ gsd_xrandr_manager_start (GsdXrandrManag
 
         g_signal_connect (manager->priv->rw_screen, "changed", G_CALLBACK (on_randr_event), manager);
 
+#if HAVE_UPOWER_GLIB
         manager->priv->upower_client = up_client_new ();
         manager->priv->laptop_lid_is_closed = up_client_get_lid_is_closed (manager->priv->upower_client);
         g_signal_connect (manager->priv->upower_client, "changed",
                           G_CALLBACK (power_client_changed_cb), manager);
+#endif
 
         log_msg ("State of screen at startup:\n");
         log_screen (manager->priv->rw_screen);
@@ -2025,11 +2038,13 @@ gsd_xrandr_manager_stop (GsdXrandrManage
                 manager->priv->rw_screen = NULL;
         }
 
+#if HAVE_UPOWER_GLIB
         if (manager->priv->upower_client != NULL) {
                 g_signal_handlers_disconnect_by_data (manager->priv->upower_client, manager);
                 g_object_unref (manager->priv->upower_client);
                 manager->priv->upower_client = NULL;
         }
+#endif
 
         if (manager->priv->introspection_data) {
                 g_dbus_node_info_unref (manager->priv->introspection_data);
