--- gnome-panel-3.4.1/configure.ac-orig	2012-05-09 11:49:37.101576007 -0500
+++ gnome-panel-3.4.1/configure.ac	2012-05-09 11:50:23.272049608 -0500
@@ -149,10 +149,22 @@ PKG_CHECK_MODULES(TZ, gio-2.0 >= $GLIB_R
 AC_SUBST(TZ_CFLAGS)
 AC_SUBST(TZ_LIBS)
 
-PKG_CHECK_MODULES(CLOCK, pango >= $PANGO_REQUIRED gtk+-3.0 >= $GTK_REQUIRED glib-2.0 >= $GLIB_REQUIRED gio-2.0 >= $GLIB_REQUIRED $LIBECAL_REQUIREMENT librsvg-2.0 gweather-3.0 >= $GWEATHER_REQUIRED gnome-desktop-3.0 >= $LIBGNOME_DESKTOP_REQUIRED polkit-gobject-1)
+PKG_CHECK_MODULES(CLOCK, pango >= $PANGO_REQUIRED gtk+-3.0 >= $GTK_REQUIRED glib-2.0 >= $GLIB_REQUIRED gio-2.0 >= $GLIB_REQUIRED $LIBECAL_REQUIREMENT librsvg-2.0 gweather-3.0 >= $GWEATHER_REQUIRED gnome-desktop-3.0 >= $LIBGNOME_DESKTOP_REQUIRED)
 AC_SUBST(CLOCK_CFLAGS)
 AC_SUBST(CLOCK_LIBS)
 
+PKG_CHECK_MODULES(POLKIT,
+        polkit-gobject-1,
+        have_polkit=yes,
+        have_polkit=no)
+if test "x$have_polkit" = "xyes" ; then
+  AC_DEFINE(HAVE_POLKIT, [], [Define if we have polkit])
+fi
+AM_CONDITIONAL(HAVE_POLKIT, test x$have_polkit = xyes)
+AC_SUBST(HAVE_POLKIT)
+AC_SUBST(POLKIT_CFLAGS)
+AC_SUBST(POLKIT_LIBS)
+
 DBUS_REQUIRED=1.1.2
 NETWORK_MANAGER_REQUIRED=0.6
  
--- gnome-panel-3.4.1/applets/clock/set-timezone.c-orig	2012-05-09 13:00:17.369915353 -0500
+++ gnome-panel-3.4.1/applets/clock/set-timezone.c	2012-05-09 13:01:42.624255743 -0500
@@ -23,7 +23,9 @@
 #endif
 
 #include <gio/gio.h>
+#ifdef HAVE_POLKIT
 #include <polkit/polkit.h>
+#endif
 
 #include "set-timezone.h"
 
@@ -62,6 +64,7 @@ get_system_bus (GError **error)
 static int
 can_set (Cache *cache, const gchar *method_name)
 {
+#if HAVE_POLKIT
   guint64 now = g_get_monotonic_time ();
 
   if (now - cache->stamp > (CACHE_VALIDITY_SEC * 1000000))
@@ -102,6 +105,9 @@ can_set (Cache *cache, const gchar *meth
     }
 
   return cache->value;
+#else
+  return 0;
+#endif
 }
 
 gint
