diff -urN gnome-session-2.27.5/data/gnome-session.schemas.in.in ../SUNWgnome-session-2.26.1.p10/gnome-session-2.27.5/data/gnome-session.schemas.in.in
--- gnome-session-2.27.5/data/gnome-session.schemas.in.in	2009-07-29 01:36:07.000000000 +0100
+++ ../SUNWgnome-session-2.26.1.p10/gnome-session-2.27.5/data/gnome-session.schemas.in.in	2009-08-06 18:07:09.121100193 +0100
@@ -7,7 +7,7 @@
          <applyto>/apps/gnome-session/options/show_splash_screen</applyto>
          <owner>gnome</owner>
          <type>bool</type>
-         <default>false</default>
+         <default>true</default>
          <locale name="C">
             <short>Show the splash screen</short>
             <long>Show the splash screen when the session starts up</long>
