--- gnome-settings-daemon-3.1.5/configure.ac-orig	2011-08-24 09:18:29.710425446 -0500
+++ gnome-settings-daemon-3.1.5/configure.ac	2011-08-24 09:18:58.492535062 -0500
@@ -81,7 +81,7 @@ PKG_CHECK_MODULES(BACKLIGHT_HELPER,
         gio-2.0 >= $GIO_REQUIRED_VERSION
 )
 
-GSD_PLUGIN_LDFLAGS="-export_dynamic -module -avoid-version -no-undefined"
+GSD_PLUGIN_LDFLAGS="-export_dynamic -module -avoid-version -no-undefined -Wl,-M,../../mapfile-gsd"
 case $host_os in
   darwin*)
     GSD_PLUGIN_LDFLAGS="${GSD_PLUGIN_LDFLAGS} -Wl,-bundle_loader,\$(top_builddir)/gnome-settings-daemon/gnome-settings-daemon"
--- /dev/null	2011-08-24 09:21:19.000000000 -0500
+++ gnome-settings-daemon-3.1.5/mapfile-gsd	2011-08-23 16:00:23.238440669 -0500
@@ -0,0 +1,30 @@
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+# Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
+# Use is subject to license terms.
+#
+# ident "@(#)mapfile-libsqlite3 1.4     09/06/05 SMI"
+#
+# Defines the public interface to SQLite3
+#                    
+{
+	global: 
+                gnome_settings_plugin_get_type = FUNCTION extern;
+};
