--- gst-plugins-base-0.10.35/gst/videotestsrc/Makefile.am-orig	2011-10-01 22:51:33.341823545 -0500
+++ gst-plugins-base-0.10.35/gst/videotestsrc/Makefile.am	2011-10-01 22:52:30.926832198 -0500
@@ -19,7 +19,7 @@ noinst_PROGRAMS = generate_sine_table
 
 generate_sine_table_SOURCES = generate_sine_table.c
 generate_sine_table_CFLAGS = $(GST_CFLAGS)
-generate_sine_table_LDADD = $(LIBM)
+generate_sine_table_LDADD = $(LIBM) $(GLIB_LIBS)
 
 Android.mk: Makefile.am $(BUILT_SOURCES)
 	androgenizer \
