diff -ruN gtk-vnc-0.3.10.orig/plugin/Makefile.am gtk-vnc-0.3.10/plugin/Makefile.am
--- gtk-vnc-0.3.10.orig/plugin/Makefile.am	2009-08-28 17:31:55.000000000 +0100
+++ gtk-vnc-0.3.10/plugin/Makefile.am	2012-05-14 23:49:22.856785630 +0100
@@ -12,7 +12,7 @@
 	-module -avoid-version
 gtk_vnc_plugin_la_CFLAGS = \
 	-I$(top_srcdir)/src @GTK_CFLAGS@ \
-	@FIREFOX_PLUGIN_CFLAGS@ @WARNING_CFLAGS@ @WERROR_CFLAGS@
+	@FIREFOX_PLUGIN_CFLAGS@ @WARNING_CFLAGS@
 
 all-local: gtk-vnc-plugin.so
 
