--- gtk+-3.2.0/gtk/gtkfontbutton.c-orig	2011-09-30 19:28:42.579503405 -0500
+++ gtk+-3.2.0/gtk/gtkfontbutton.c	2011-09-30 19:29:15.394325389 -0500
@@ -275,9 +275,9 @@ gtk_font_button_set_show_preview_entry (
   GtkFontButtonPrivate *priv = font_button->priv;
 
   if (priv->font_dialog)
-    return gtk_font_chooser_set_show_preview_entry (GTK_FONT_CHOOSER (priv->font_dialog), show);
-
-  priv->show_preview_entry = show != FALSE;
+    gtk_font_chooser_set_show_preview_entry (GTK_FONT_CHOOSER (priv->font_dialog), show);
+  else
+    priv->show_preview_entry = show != FALSE;
 }
 
 static PangoFontFamily *
