diff -urN gtkam-0.1.17/Makefile.am ../SUNWgnome-camera-2.30.0.hacked/gtkam-0.1.17/Makefile.am
--- gtkam-0.1.17/Makefile.am	2010-01-31 15:13:09.000000000 +0000
+++ ../SUNWgnome-camera-2.30.0.hacked/gtkam-0.1.17/Makefile.am	2011-05-30 16:24:34.837048554 +0000
@@ -2,7 +2,7 @@
 # Process this file with automake to produce Makefile.in
 ########################################################################
 
-SUBDIRS = m4m art doc help omf-install packaging pixmaps po src test
+SUBDIRS = m4m art doc help packaging pixmaps po src test
 
 ACLOCAL_AMFLAGS = -I m4m
 
diff -urN gtkam-0.1.17/configure.ac ../SUNWgnome-camera-2.30.0.hacked/gtkam-0.1.17/configure.ac
--- gtkam-0.1.17/configure.ac	2010-01-31 16:14:28.000000000 +0000
+++ ../SUNWgnome-camera-2.30.0.hacked/gtkam-0.1.17/configure.ac	2011-05-30 16:19:06.697299439 +0000
@@ -260,8 +260,6 @@
 src/Makefile
 test/Makefile
 doc/Makefile
-omf-install/Makefile
-omf-install/gtkam-C.omf
 help/Makefile
 help/C/Makefile
 help/ru/Makefile
