diff -urN gtkmm-2.17.2/gtk/gtkmm/notebook.cc ../SUNWgtkmm-2.17.2.hacked/gtkmm-2.17.2/gtk/gtkmm/notebook.cc
--- gtkmm-2.17.2/gtk/gtkmm/notebook.cc  2009-07-13 18:09:52.000000000 +0100
+++ ../SUNWgtkmm-2.17.2.hacked/gtkmm-2.17.2/gtk/gtkmm/notebook.cc       2009-07-24 14:34:34.608735637 +0100
@@ -284,8 +284,9 @@
 {
   if (num < 0)
     return end();
-  iterator i = begin();
-  std::advance(i, num);
+  int j;
+  iterator i;
+  for (i = begin(), j = 0; i != end(), j < num; ++i, ++j);
   return i;
 }
 
diff -urN gtkmm-2.17.2/gtk/src/notebook.ccg ../SUNWgtkmm-2.17.2.hacked/gtkmm-2.17.2/gtk/src/notebook.ccg
--- gtkmm-2.17.2/gtk/src/notebook.ccg   2009-07-13 16:31:30.000000000 +0100
+++ ../SUNWgtkmm-2.17.2.hacked/gtkmm-2.17.2/gtk/src/notebook.ccg        2009-07-24 13:33:51.917473013 +0100
@@ -277,8 +277,9 @@
 {
   if (num < 0)
     return end();
-  iterator i = begin();
-  std::advance(i, num);
+  int j;
+  iterator i;
+  for (i = begin(), j = 0; i != end(), j < num; ++i, ++j);
   return i;
 }
