--- gc-7.1/doc/gc.man.old	2008年  7月 31日 四 17:12:58
+++ gc-7.1/doc/gc.man	2008年  7月 31日 四 17:13:34
@@ -1,4 +1,7 @@
+'\" te
 .TH GC_MALLOC 1L "2 October 2003"
+...\" LSARC 2008/068 libgc 
+
 .SH NAME
 GC_malloc, GC_malloc_atomic, GC_free, GC_realloc, GC_enable_incremental, GC_register_finalizer, GC_malloc_ignore_off_page, GC_malloc_atomic_ignore_off_page, GC_set_warn_proc \- Garbage collecting malloc replacement
 .SH SYNOPSIS
@@ -95,3 +98,23 @@
 .SH AUTHOR
 Hans-J. Boehm (Hans.Boehm@hp.com).
 Some of the code was written by others, most notably Alan Demers.
+
+.SH "ATTRIBUTES"
+.PP
+See \fBattributes\fR(5)
+for descriptions of the following attributes:
+.sp
+.TS
+tab(^G) box;
+cw(2.75i) |cw(2.75i)
+lw(2.75i) |lw(2.75i)
+.
+ATTRIBUTE TYPE^ATTRIBUTE VALUE
+_
+Availability^SUNWlibgc
+_
+Interface stability^Uncommitted
+
+.TE
+.sp
+
