--- libgnomekbd-3.4.0.2/test/Makefile.am-orig	2012-05-04 12:29:53.888573660 -0500
+++ libgnomekbd-3.4.0.2/test/Makefile.am	2012-05-04 12:29:59.997036636 -0500
@@ -2,7 +2,7 @@ noinst_PROGRAMS = gkbd-indicator-test \
                   gkbd-keyboard-drawing-test \
                   gkbd-status-test
 
-CFLAGS_common = $(WARN_CFLAGS) -I$(top_srcdir) -Wall \
+CFLAGS_common = $(WARN_CFLAGS) -I$(top_srcdir) \
 	$(GTK_CFLAGS) \
 	$(LIBXKLAVIER_CFLAGS) \
 	-DGNOMELOCALEDIR="\"$(datadir)/locale\""
