diff -rup SUNWpango-1.25.5/i386/pango-1.25.5/configure.in pango-1.25.5/configure.in
--- SUNWpango-1.25.5/i386/pango-1.25.5/configure.in	2009-08-27 14:45:12.931222570 +0200
+++ pango-1.25.5/configure.in	2009-08-27 14:46:25.400555664 +0200
@@ -143,9 +143,11 @@ AC_CHECK_FUNCS(mprotect sysconf getpages
 AC_CHECK_HEADERS(unistd.h sys/mman.h)
 
 # Make sure we don't link to libstdc++ (needs de-gcc-fication)
-CXXFLAGS="$CXXFLAGS -fno-rtti -fno-exceptions"
-
-
+if test "$GCC" = yes; then
+    CXXFLAGS="$CXXFLAGS -fno-rtti -fno-exceptions"
+  
+    
+fi
 #
 # Win32 stuff
 #
