--- libtool-2.2.6b/libltdl/ltdl.h-orig	2011-09-27 17:30:52.685894801 -0500
+++ libtool-2.2.6b/libltdl/ltdl.h	2011-09-27 17:45:22.205222161 -0500
@@ -101,7 +101,7 @@ LT_SCOPE int	lt_dlpreload_default (const
 LT_SCOPE int	lt_dlpreload_open    (const char *originator,
 				      lt_dlpreload_callback_func *func);
 
-#define lt_preloaded_symbols	lt__PROGRAM__LTX_preloaded_symbols
+#define lt_preloaded_symbols	lt_libltdl_LTX_preloaded_symbols
 #define LTDL_SET_PRELOADED_SYMBOLS() 			LT_STMT_START{	\
 	extern const lt_dlsymlist lt_preloaded_symbols[];		\
 	lt_dlpreload_default(lt_preloaded_symbols);			\
