--- pulseaudio-1.1/src/daemon/default.pa.in-orig	2012-05-17 01:30:56.827201316 -0500
+++ pulseaudio-1.1/src/daemon/default.pa.in	2012-05-17 01:31:12.110991968 -0500
@@ -52,7 +52,7 @@ ifelse(@HAVE_ALSA@, 1, [dnl
 #load-module module-alsa-source device=hw:1,0
 ])dnl
 ifelse(@HAVE_OSS_OUTPUT@, 1, [dnl
-#load-module module-oss device="/dev/dsp" sink_name=output source_name=input
+load-module module-oss device="/dev/dsp" sink_name=output source_name=input
 #load-module module-oss-mmap device="/dev/dsp" sink_name=output source_name=input
 ])dnl
 ifelse(@HAVE_WAVEOUT@, 1, [dnl
