--- tsoljdslabel-0.6.4/configure.in.orig	2009-02-12 22:59:56.222687000 +0900
+++ tsoljdslabel-0.6.4/configure.in	2009-02-12 23:00:35.088966000 +0900
@@ -26,7 +26,7 @@ AC_ARG_WITH(dtstart, 
 if test "x$use_dtstart" = "xyes" ; then
 	START_COMMAND="/usr/bin/dtstart tjds"
 else
-	START_COMMAND="/usr/dt/config/Xsession.tjds"
+	START_COMMAND="/usr/bin/gnome-session --trusted-session"
 fi
 AC_SUBST(START_COMMAND)
 
