--- tsoljdsdevmgr-0.6.2/src/Makefile.am.orig	2008-07-10 16:05:23.633126300 +0100
+++ tsoljdsdevmgr-0.6.2/src/Makefile.am	2008-07-10 16:05:30.071778500 +0100
@@ -1,6 +1,6 @@
 INCLUDES = -I. -I$(srcdir) -I/usr/openwin/include/X11/extenstions 	\
 	-DGNOMELOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\" 		\
-	-DG_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED 		\
+	-DG_DISABLE_DEPRECATED \
 	-DGNOME_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED 		\
 	-DGNOME_ICONDIR=\"$(prefix)/share/pixmaps\"			\
 	$(TSOLJDSDEVMGR_CFLAGS)				
